com.onesignal.unity.android/Runtime/AndroidJavaProxy.cs

```diff
@@ -0,0 +1,96 @@
+using Laters;
+using UnityEngine;
+using System;
+using System.Runtime.CompilerServices;
+using System.Threading.Tasks;
+
+namespace OneSignalSDKNew {
+    public abstract class OneSignalAndroidJavaProxy : AndroidJavaProxy {
+        private const string SDKPackage = "com.onesignal";
+        private const string SDKClassName = "OneSignal";
+        private const string QualifiedSDKClass = SDKPackage + "." + SDKClassName;
+
+        protected OneSignalAndroidJavaProxy(string listenerClassName, bool innerClass = false)
+            : base(innerClass
+                ? QualifiedSDKClass + "$" + listenerClassName // com.onesignal.OneSignal$[name], true
```

Comment:
I think this logic can be cleaned up at some point.  There are no longer inner classes to `com.onesignal.OneSignal`.  Consider in the next PR we get rid of `OneSignalAndroidJavaProxy` and just use `AndroidJavaProxy` with fully qualified names.  Alternatively we get rid of `innerClass` and keep the abstraction where we assume every class starts with "com.onesignal".

Same comment for `OneSignalAwaitableAndroidJavaProxy`

Reply:
I think I'm going to keep the "com.onesignal" abstraction. Going to cleanup and remove the inner class in another PR